R/readWQPdata.R

```diff
@@ -96,12 +96,25 @@
 #' startDate <- as.Date("2013-01-01")
 #' nutrientDaneCounty <- readWQPdata(countycode="US:55:025",startDate=startDate,
 #'                         characteristicType="Nutrient")
+#' secchi.names = c("Depth, Secchi disk depth", 
+#'                  "Depth, Secchi disk depth (choice list)", 
+#'                  "Secchi Reading Condition (choice list)", 
+#'                  "Secchi depth", 
+#'                  "Water transparency, Secchi disc")
+#' args <- list('startDateLo' = startDate, 
+#'              'startDateHi' = "2013-12-31", 
+#'               statecode="WI", 
+#'               characteristicName=secchi.names)
 #' 
-#'                         
+#' wqp.data <- readWQPdata(args)                  
 #' }
 readWQPdata <- function(..., zip=FALSE, querySummary=FALSE){
```

Comment:
I link this would be useful if we can hammer out some of the details and figure out what the expected pattern should be. 

Questions:
1) do we accept more than one list in `...`?
2) can you override arguments outside of the list? (e.g., what happens when I specify a value for `zip` in the list?
3) can non-list args be included in `...` and what happens if not? (i.e., if we have one list in `...` and the rest args, does it fail gracefully and let you know why?)

I guess I was thinking this would have a list, not be all of none on `...` being lists. A couple of challenges that we might run into with this pattern:
```r
args <- list('startDateLo' = startDate, 
              'startDateHi' = "2013-12-31")
wqp.data <- readWQPdata(args, statecode="WI", characteristicName=secchi.names)
```
or
```r
args <- list('startDateLo' = startDate, 
              'startDateHi' = "2013-12-31", zip = FALSE)
wqp.data <- readWQPdata(args) 
```
The main use-case I would have is having all of the args in a list and using `do.call`, but I think `zip` and `querySummary` would be blocked from that pattern in this implementation.